### PR TITLE
Add support for 4 date based N1QL functions

### PR DESCRIPTION
**Pull request by borrrden**
- DATE_DIFF_STR
- DATE_DIFF_MILLIS
- DATE_ADD_STR
- DATE_ADD_MILLIS

### CODE REVIEW

File: LiteCore/Query/SQLiteN1QLFunctions.cc [hunk 1]
```diff
@@ -813,28 +837,223 @@ namespace litecore {
 
 #pragma mark - DATES:
 
+    inline int64_t getQuarter(const DateTime& t) {
```
Comment (snej) [inline LiteCore/Query/SQLiteN1QLFunctions.cc]: This is a lot of code being added — could you modularize this by creating new source/header files for the date-time logic, and then having just the SQLite glue code in here?

File: LiteCore/Query/SQLiteN1QLFunctions.cc [hunk 2]
```diff
@@ -813,28 +837,223 @@ namespace litecore {
...
+        return (t.M + 2) / 3;
+    }
+
+    inline int64_t leapYearsWithin(int64_t year) {
```
Comment (snej) [inline LiteCore/Query/SQLiteN1QLFunctions.cc]: What's the source of these algorithms/formulas? Did they come from existing working code?

Comment (borrrden) [reply LiteCore/Query/SQLiteN1QLFunctions.cc]: I think the source was the N1QL go code from the server team.

**Comment by borrrden** [comment]
This PR appears to be unimportant.  I'll leave the branch around.

**Pull request closed without merging**